### PR TITLE
Fix #45644: Error message for invalid errorlog format option is incorrect

### LINKED ISSUE #45644: Error message for invalid errorlog format option is incorrect
**Version Used**: 

3.6.0-4.20251.5 (910223b6)

**Steps to Reproduce**:

1. Issue the command

    ```
    csc Program.cs /errorlog:Program.sarif,version=2.1.0
    ```

**Expected Behavior**:

The error message should be

```
error CS2046: Command-line syntax error: 'Program.sarif,version=2.1.0' is not a valid value for the '/errorlog:' option.
The value must be of the form '<file>[,version={1|1.0|2|2.1}]'.
```

**Actual Behavior**:

The error message is

```
error CS2046: Command-line syntax error: 'Program.sarif,version=2.1.0' is not a valid value for the '/errorlog:' option.
The value must be of the form '<file>[,version={1|1.0|1.0.0|2|2.1|2.1.0}]'.
```

... even though `1.0.0` and `2.1.0` are not accepted by the command line parser.

The same thing happens with VB.

### PR DESCRIPTION
**Pull request by lgolding**
@michaelcfanning @paulomorgado FYI

Fixes https://github.com/dotnet/roslyn/issues/45644

### CODE REVIEW

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -4060,7 +4060,7 @@ public void ParseErrorLog()
             const string InvalidSarifVersion = @"C:\MyFolder\MyBinary.xml,version=42";
             parsedArgs = DefaultParse(new[] { $"/errorlog:{InvalidSarifVersion}", "a.cs" }, baseDirectory);
             parsedArgs.Errors.Verify(
-                // error CS2046: Command-line syntax error: 'C:\MyFolder\MyBinary.xml,version=42' is not a valid value for the '/errorlog:' option. The value must be of the form '<file>[,version={1|1.0|1.0.0|2|2.1|2.1.0}]'.
+                // error CS2046: Command-line syntax error: 'C:\MyFolder\MyBinary.xml,version=42' is not a valid value for the '/errorlog:' option. The value must be of the form '<file>[,version={1|1.0|2|2.1}]'.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:4063]: >// error CS2046: Command-line syntax error: 'C:\MyFolder\MyBinary.xml,version=42' is not a valid value for the '/errorlog:' option. The value must be of the form '<file>[,version={1|1.0|2|2.1}]'. [](start = 16, length = 195)

Comments do not verify anything. Please add tests that demonstrate that the options that you removed from the list are indeed errors. Please also add tests for VB

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

**Comment by jcouv** [comment]
Marked PR as draft at the moment, to declutter our code review backlog. Please re-activate or ping when ready for another look. Thanks

**Comment by eddynaka** [comment]
@AlekseyTs , we can close this, since we merged my PR. thank you.

**Pull request closed without merging**